docs/source/generics.rst

```diff
@@ -553,16 +552,17 @@ Here's a complete example of a function decorator:
 
 .. code-block:: python
 
-   from typing import Any, Callable, TypeVar, cast
+   from typing import Callable, ParamSpec, TypeVar
```

Comment:
I think that we should include both examples. Because both techniques are quite common. This way we can clearly show why `ParamSpec` is better.

What do you think?

Reply:
@sobolevn Check it out now. I've re-introduced the non-ParamSpec example, and also added an example where we change the return type (to show where ParamSpec is actually useful beyond "avoiding casts" type pure-ism).